samples/drivers/drv2605/src/main.c

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2024 Cirrus Logic, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/haptics.h>
+#include <zephyr/drivers/haptics/drv2605.h>
+#include <zephyr/sys/util.h>
+#include <sys/_stdint.h>
+
+#define LOG_LEVEL 4
+
+LOG_MODULE_REGISTER(main);
+
+static struct drv2605_rom_data rom_data = {
```

Comment:
Would it make sense at all for the driver to have a default ROM? It kind of bothers me that the first sample being introduced for the new subsystem can't be 100% generic, but maybe that's just the way it is 
.. so not directly a request for change, just asking in case there is a way around this :)
Also wondering whether ROM information could be provided as devicetree overlay? (i.e. if it's "ROM" one can probably argue it's very much hardware bound and not really dynamic configuration?)

Reply:
Hey @kartben, the ROM refers to the bank of licensed waveforms built-in on the DRV2605. By default the sequencer queue on the DRV2605 is empty, otherwise I would have a generic sample app this chip could make use of where we play the default waveform. So unfortunately I think that's just the way it is.

We could introduce configuring the sequencer and the related controls to a default condition in the devicetree, but I feel that this will be set at runtime most of the time. If you are using ROM as a playback source there are various waveforms in each bank that could be used for different feedback events. Alternatively the other playback sources supported by the DRV2605 rely on some external stimulus (PWM, analog signal, analog audio, streamed values) to be applied at runtime.